pkg/user/service.go

```diff
@@ -11,20 +11,19 @@ import (
 	"golang.org/x/crypto/scrypt"
 )
 
-type Service interface {
-	SignUp(email string, password string) (*model.User, error)
-	SignIn(email string, password string) (*model.User, error)
-	FindById(id uint) (*model.User, error)
-	FindByEmail(email string) (*model.User, error)
-	FindOrCreate(email string, password string) (*model.User, error)
+func NewService(repository userRepository) *service {
+	return &service{repository}
 }
 
-type service struct {
-	repository Repository
+type userRepository interface {
+	create(user *model.User) error
+	findByEmail(email string) (*model.User, error)
+	FindById(id uint) (*model.User, error)
```

Comment:
FindById should probably also be small caps, right?

Reply:
If I unexport FindById I get the following error in main.go

    Cannot use 'usrRepository' (type *repository) as the type userRepository Type does not implement 'userRepository' as some methods are missing: FindById(id uint) (*model.User, error)

I'm considering if we should pass the service rather than the repository but I suspect we'll have the same problem.